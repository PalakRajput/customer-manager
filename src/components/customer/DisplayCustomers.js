import React, { useEffect } from 'react'
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux/';
import { deleteCustomer, getCustomerAction } from '../../redux/actions/CustomerActions';

const DisplayCustomers = () => {
    const dispatch = useDispatch();
    const customers = useSelector((state) => state.customer.customers);
    //const successMessage = useSelector(state => state.customer.successMessage)
    useEffect(() => {
            dispatch(getCustomerAction())
    },[])
    const handleDeleteClick = (id) => {
        dispatch(deleteCustomer(id))
    }
    return (
        <>
            <div style={{margin:'20px'}}>
                <div className='row'>

                    {customers.length !==0 ? (customers.map(customer => (
                        <div className='col md-2' key={customer.id}>
                            <div className='card' style={{ boxShadow: '10px 10px 5px gray' }}>
                                <div className='card-body' style={{ background: 'linear-gradient(270deg, #c5e6ec 0%, rgb(248 212 248) 100%)' }}>
                                    <div className='card-title' style={{ fontWeight: 'bold' }}><Link to='/' state={{customer}}>{customer.fname + " " + customer.lname}</Link></div>
                                    {customer.city}
                                    <br />
                                    <Link to={`/orders/${customer.id}`} style={{ fontWeight: 'bold' }}>View {customer.ordersList.length} Orders</Link>
                                </div>
                                <div className='card-footer'>
                                    <button className='btn btn-danger btn-small' onClick={() => handleDeleteClick(customer.id)}>Delete</button>
                                </div>
                            </div>
                        </div>
                    ))
                    ) : (
                        <div>No Customers to Display. Please add customers</div>
                    )
                }


                </div>
            </div>
        </>
    )
}
export default DisplayCustomers;