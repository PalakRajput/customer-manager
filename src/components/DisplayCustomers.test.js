import { render, screen } from '@testing-library/react'
import React from 'react'
import DisplayCustomers from './DisplayCustomers'

describe("Display Customers", () => {
    it (" should render correctly ", () => {
           const {asFragment} =  render(<DisplayCustomers />)
           expect(asFragment(<DisplayCustomers />)).toMatchSnapshot();
    })

    it (" should have h2 with text customers", () => {
        render(<DisplayCustomers />)
        //const heading = 
        screen.getByText(/customers/i)
        //expect(heading).toBeInTheDocument();
    })
})