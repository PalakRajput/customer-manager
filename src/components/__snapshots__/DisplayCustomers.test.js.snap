// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Customers  should render correctly  1`] = `
<DocumentFragment>
  <h2>
    Customers
  </h2>
</DocumentFragment>
`;
